src/main/java/org/mafagafogigante/dungeon/stats/ExplorationStatistics.java

```diff
@@ -38,9 +44,32 @@ public void createEntryIfNotExists(Point point, Id locationId) {
    *
    * @param point the visited Point
    */
-  public void addVisit(Point point, Id locationId) {
-    createEntryIfNotExists(point, locationId);
+  public void addVisit(Point point, Id locationId, Date discoveredDate) {
+    createEntryIfNotExists(point, locationId, discoveredDate);
     entries.get(point).addVisit();
+    addDistinctLocationByPartOfDay(point);
+  }
+
+  /**
+   * Records a distinct location corresponding to partOfDay.
+   *
+   * @param point the visited Point
+   */
+  public void addDistinctLocationByPartOfDay(Point point) {
+    boolean distinct = true;
+    for (ExplorationStatisticsEntry location : distinctLocations.values()) {
+      if (location.getLocationId() == entries.get(point).getLocationId() &&
+          PartOfDay.getCorrespondingConstant(
```

Comment:
You can make this more readable by introducing local variables.

Having a multi-line expression is not ideal.

Reply:
Is something like this better?

```
    boolean distinct = true;
    boolean sameLocation;
    boolean samePartOfDay;
    ExplorationStatisticsEntry entry = entries.get(point);
    for (ExplorationStatisticsEntry location : distinctLocations.values()) {
      sameLocation = location.getLocationId() == entry.getLocationId();
      samePartOfDay = PartOfDay.getCorrespondingConstant(entry.getDiscoveredDate()).getName().getSingular()
          .equalsIgnoreCase(PartOfDay.getCorrespondingConstant(location.getDiscoveredDate()).toString());
      if (sameLocation && samePartOfDay){
        distinct = false;
        break;
      }
    }
```

Reply:
Note: For some reason I have to use function `equalsIgnoreCase` instead of  `==`. When I print it out to the console the location names seems to be the same but the expression returns false.

Reply:
Actually maybe this one is better:

```
    boolean distinct = true;
    String entryLocationName;
    String distinctLocationName;
    boolean matchingLocation;
    for (ExplorationStatisticsEntry location : distinctLocations.values()) {
      entryLocationName = PartOfDay.getCorrespondingConstant(entries.get(point).getDiscoveredDate()).getName().getSingular();
      distinctLocationName = PartOfDay.getCorrespondingConstant(location.getDiscoveredDate()).toString();
      matchingLocation = location.getLocationId() == entries.get(point).getLocationId();
      if (matchingLocation && entryLocationName.equalsIgnoreCase(distinctLocationName)){
        distinct = false;
        break;
      }
    }
```

Reply:
Yes, you are getting what I meant.

However, it is better to only declare the variables when you need them. Having ```boolean matchingLocation``` for a few lines before using it will introduce an undefined variable with no need for it. In general, only declare variables when you cannot defer it any further.

Reply:
So like this:
```java
for (ExplorationStatisticsEntry location : distinctLocations.values()) {
      PartOfDay day = PartOfDay.getCorrespondingConstant(entries.get(point).getDiscoveredDate());
      String entryName = day.getName().getSingular();
      String distinctName = PartOfDay.getCorrespondingConstant(location.getDiscoveredDate()).toString();
      if (location.getLocationId() == entries.get(point).getLocationId() && entryName.equalsIgnoreCase(distinctName)) {
        distinct = false;
        break;
      }
    }
```

Reply:
Looks good.